File: mindsurf-voice-ai/src/services/appInfo.ts
import { invoke } from "@tauri-apps/api/core";

import type { AppInfo, CommandResult } from "../types/app";

export async function getAppInfo(): Promise<CommandResult<AppInfo>> {
  try {
    return await invoke<CommandResult<AppInfo>>("get_app_info");
  } catch {
    return {
      ok: false,
      error: {
        code: "app_info_unavailable",
        message: "无法读取应用构建信息",
        recoverable: true,
      },
    };
  }
}

File: mindsurf-voice-ai/src-tauri/src/commands/app_info.rs
use serde::Serialize;

use crate::error::{AppError, CommandResult};

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct AppInfo {
    version: &'static str,
    platform: &'static str,
    arch: &'static str,
    build_profile: &'static str,
}

#[tauri::command]
pub fn get_app_info() -> CommandResult<AppInfo> {
    let version = env!("CARGO_PKG_VERSION");

    if version.trim().is_empty() {
        return CommandResult::failure(AppError::new(
            "app_info_unavailable",
            "package version is missing",
            false,
        ));
    }

    CommandResult::success(AppInfo {
        version,
        platform: std::env::consts::OS,
        arch: std::env::consts::ARCH,
        build_profile: if cfg!(debug_assertions) {
            "debug"
        } else {
            "release"
        },
    })
}

#[cfg(test)]
mod tests {
    use super::get_app_info;

    #[test]
    fn app_info_uses_package_version() {
        let result = get_app_info();
        let value = serde_json::to_value(result).expect("app info should serialize");

        assert_eq!(value["ok"], true);
        assert_eq!(value["data"]["version"], env!("CARGO_PKG_VERSION"));
    }
}
